### PR TITLE
Datamining blog for the demo hawkular alpha 9

### PR DESCRIPTION
**Pull request by pavolloffay**
@Jiri-Kremser @pilhuhn could you please review? 

The presentation is here:
https://docs.google.com/presentation/d/1VPvKatJzBN3yQNjNyC-utBAzpWHMpOwNlfmrXbC5i9U/edit?usp=sharing

### CODE REVIEW

File: src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
+= Hawkular Data Mining - Predictive Charts
+Pavol Loffay
+2016-1-20
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, data mining
+
+I would like to present you functionality of Hawkular Data Mining module.
+
+Hawkular Data Mining module can be deployed in Hawkular and predict all metrics. Predicted values are
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc]: s/predict all/and can be used to predict future values of metrics/ 

?

File: src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+automatically sent to Alerts where conditions are evaluated and an alert can be triggered.
+
+=== Time series models
+At the beginning we link:http://www.hawkular.org/blog/2015/10/24/hawkular-datamining.html[started] with linear
+regression in Apache Spark. This simple model was a good start. Now we use Holt's linear trend method which is
+a variant of exponential smoothing with linear trend. For forecasting an alert is important to model a trend. ARIMA models
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc]: Explain ARIMA - and/or add a reference to e.g. wikipedia

File: src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+were also proposed, however they require stationary time series and periodical recalculation of parameters would
+have to be done. Which requires all historical data in place. For seasonal time series we plan to sue Holt Winters
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc]: s/sue/use/

File: src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ $ curl -ivX POST -H "Content-Type: application/json" 'http://jdoe:password@localhost:8080/hawkular/inventory/tenants/relationships' -d '{
+     "name": "__inPrediction",
+     "source": "'$tenant'",
+     "target": "'$tenant'",
+     "properties": {"predictionInterval": 150}
+ }'
+--
+
+Now we can query predictions for any time in the future. Of course we cannot get reasonable result for predictions
+of further future.
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc]: Explain what this means.

File: src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+.Get five step ahead predictions for Heap Used metric, link:https://gist.github.com/pavolloffay/5c0a7aea318818f59293[helper functions]
```
Comment (pilhuhn) [inline src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc:57]: What is a 'step' ?

File: src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+.Enable predictions for all metrics of given tenant.
+--
+ $ tenant=$(curl -s GET 'http://jdoe:password@localhost:8080/hawkular/inventory/tenant'| grep --color=never -oP 'path" : "\K/t;[0-9a-z\-]+')
+ $ curl -ivX POST -H "Content-Type: application/json" 'http://jdoe:password@localhost:8080/hawkular/inventory/tenants/relationships' -d '{
+     "name": "__inPrediction",
+     "source": "'$tenant'",
+     "target": "'$tenant'",
+     "properties": {"predictionInterval": 150}
+ }'
```
Comment (pavolloffay) [inline src/main/jbake/content/blog/2016/01/20/hawkular-datamining-demo-hwk-alpha9.adoc:48]: TODO I will rename it to forecastingHorizon as this collide with confidence interval which is in time series theory called  also prediction interval...

**Comment by pavolloffay** [comment]
Thanks @pilhuhn

**Comment by jkremser** [comment]
still some minor typos like:

At the beginning we -> At the beginning, we
For forecasting an alert is important -> For alert forecasting it is important

> "Lower levels override highers." 

What does it actually mean? Can I turn the prediction on for all the metrics by adding the self-loop edge on tenant and then turn it off for some particular metric types/metrics? If I understand it well, there is only 1 type of edge and all the edge is doing, is turning on the prediction. If this is true (presence of edge = turning on), then the sentence doesn't make any sense.

> Configuration has only one property -- interval in seconds. 

It was renamed to forecast horizon, right?

up to last timestamp -> up to the last timestamp

Then predicted values are sent -> Then, the predicted values are sent

Now we can query -> Now, we can query

> The rate at which widen is in general function of selected model. For getting predictions API expects number of steps to the future. One step is equal to the interval at which given metric is collected.

This is still a little bit cryptic. What about something like:
The rate at which the confidence interval widen (~the prediction accuracy) is a function of the selected model. The prediction API takes the number of steps to the future for which it returns the predicted values. One step is equal to the interval at which given metric is collected. So the number of steps intuitively corresponds to n future measurements.

used also -> also used (but I guess both are ok)

otherwise, LGTM

**Comment by pavolloffay** [comment]
@Jiri-Kremser thanks, I think is ok now. Could you please merge?

**Pull request merged**